src/quo2/components/buttons/button/view.cljs

```diff
@@ -5,49 +5,57 @@
+    disabled boolean
+    :type   :primary/:positive:grey/:dark-grey/:outline/:ghost/
+            :danger/:black  (TODO remove :photo-bg/:blur-bg/:blur-bg-outline/:shell/:community)
+    background :photo/:blur or nil
+    :size   40 [default] /32/24/56
     :icon   true/false
-    :community-color '#FFFFFF'
-    :community-text-color '#000000'
+    :above  :icon-keyword
     :before :icon-keyword
     :after  :icon-keyword}
-
+    :customization-color keyword or hexstring 
+    :theme :light/:dark
    only icon
    [button {:icon true} :i/close-circle]"
   [_ _]
   (let [pressed-in (reagent/atom false)]
     (fn
-      [{:keys [on-press disabled type size before after above icon-secondary-no-color
+      [{:keys [on-press on-long-press disabled type background size before after above
                width customization-color theme override-background-color pressed
-               on-long-press accessibility-label icon icon-no-color style inner-style test-ID
-               blur-active? override-before-margins override-after-margins icon-size icon-container-size
-               icon-container-rounded?]
+               accessibility-label icon style inner-style
+               blur-active?]
         :or   {type                :primary
                size                40
                customization-color :primary
                blur-active?        true}}
        children]
-      (let [{:keys [icon-color icon-secondary-color background-color label-color border-color blur-type
+      (let [pressed? (or @pressed-in pressed)
+            {:keys [icon-color icon-secondary-color background-color label-color border-color blur-type
                     blur-overlay-color icon-background-color]}
-            (get-in (style/themes customization-color)
-                    [theme type])
+            (type-values/get-value-from-type {:customization-color customization-color
+                                              :background          background
+                                              :type                type
+                                              :theme               theme
+                                              :pressed?            pressed?})
             state (cond disabled                 :disabled
                         (or @pressed-in pressed) :pressed
                         :else                    :default)
             blur-state (if blur-active? :blurred :default)
-            icon-size (or icon-size (when (= 24 size) 12))
+            icon-size (when (= 24 size) 12)
             icon-secondary-color (or icon-secondary-color icon-color)]
         [rn/touchable-without-feedback
-         (merge {:test-ID             test-ID
-                 :disabled            disabled
+         (merge {:disabled            disabled
                  :accessibility-label accessibility-label
                  :on-press-in         #(reset! pressed-in true)
                  :on-press-out        #(reset! pressed-in nil)}
```

Comment:
This `merge` is adding those `on-press` and `on-long-press` when they exist, but I think if we add them, even if they are `nil` there won't be any difference.

Also, in case this is needed, we could use `cond->` + `assoc` instead of this `merge` + `when`

Reply:
thanks, I will update :)